src/components/testCaseLanding/common/import/TestCaseImportDialog.tsx

```diff
@@ -42,149 +40,164 @@ const TestCaseImportDialog = ({ open, handleClose, onImport }) => {
     setToastMessage(message);
   };
 
-  const onDrop = useCallback(async (acceptedFiles) => {
-    setErrorMessage(() => null);
-    setFileInputKey(Math.random().toString(36));
-    const importFile = acceptedFiles[0];
-    let response: ScanValidationDto;
-    try {
-      response = await testCaseService.current.scanImportFile(importFile);
-    } catch (error) {
-      showErrorToast(
-        "An error occurred while validating the import file. Please try again or reach out to the Help Desk."
-      );
-      return;
-    }
-    if (response.valid) {
-      setFile(importFile);
-      let patientBundles;
-      try {
-        patientBundles = await readImportFile(importFile);
-      } catch (error) {
-        showErrorToast(
-          "An error occurred while processing the import file. Please try to regenerate the file and re-import, or contact the Help Desk."
-        );
-        return;
-      }
-      if (patientBundles && patientBundles.length > 0) {
+  const onDrop = useCallback(async (importedZip) => {
+    removeUploadedFile();
+    setUploadingFileSpinner(true);
+    const zip = new JSZip();
+    let isValidImport = true;
+    zip.loadAsync(importedZip[0]).then((content) => {
+      Object.keys(content.files).forEach((filename) => {
```

Comment:
is this kind of reading the file?

Reply:
Yeah, using JSZip. It can help loop through all the files

Reply:
you can send the whole file as it is for scanning. Once you get a clean scan result, you can process the zip if you need individual files

Reply:
Done